### PR TITLE
Polish tracing helpers

### PR DESCRIPTION
**Pull request by LizardWizzard**
Context: comments here: https://github.com/neondatabase/neon/pull/4645

### CODE REVIEW

File: pageserver/src/tenant/timeline/span.rs [hunk 1]
```diff
@@ -1,26 +1,25 @@
+use utils::tracing_span_assert::{check_fields_present, Extractor, MultiNameExtractor};
+
 #[cfg(not(debug_assertions))]
 #[inline]
```
Comment (koivunej) [inline pageserver/src/tenant/timeline/span.rs:4]: Let's remove this as well, no need to profilirate these.
```suggestion
```

File: pageserver/src/tenant/span.rs [hunk 2]
```diff
@@ -1,24 +1,20 @@
+use utils::tracing_span_assert::{check_fields_present, MultiNameExtractor};
+
 #[cfg(not(debug_assertions))]
 #[inline]
```
Comment (koivunej) [inline pageserver/src/tenant/span.rs:4]:
```suggestion
```

**Review by koivunej — CHANGES_REQUESTED** [review]
Few more `#[inline]` removals.

Request changes for including an unexplained submodules change.

File: pageserver/src/tenant/span.rs [hunk 3]
```diff
@@ -1,24 +1,20 @@
+use utils::tracing_span_assert::{check_fields_present, MultiNameExtractor};
```
Comment (koivunej) [inline pageserver/src/tenant/span.rs:1]: Ah... clippy nags in release mode, so need either:

```suggestion
#[cfg(debug_assertions)]
use utils::tracing_span_assert::{check_fields_present, MultiNameExtractor};
```

Or wrapping the whole thing in a:

```
cfg_if::cfg_if! {
  if #[cfg(debug_assertions)] {
    use ...
    pub(crate) static TENANT_ID_EXTRACTOR: ...;
  } else {
    pub(crate) fn debug_assert_current_span_has_tenant_id() {}
  }
}
```

Probably easier to just add more annotations, it's not like this file would have any reason to change in any future.

Comment (koivunej) [reply pageserver/src/tenant/span.rs:1]: Similarly in other file.

File: pageserver/src/tenant/timeline/span.rs (hunk 1 above)
Comment (LizardWizzard) [reply pageserver/src/tenant/timeline/span.rs]: Done

File: pageserver/src/tenant/span.rs (hunk 2 above)
Comment (LizardWizzard) [reply pageserver/src/tenant/span.rs]: Done

File: pageserver/src/tenant/span.rs [hunk 4]
```diff
@@ -1,24 +1,20 @@
...
 #[inline]
 pub(crate) fn debug_assert_current_span_has_tenant_id() {}
 
 #[cfg(debug_assertions)]
-pub static TENANT_ID_EXTRACTOR: once_cell::sync::Lazy<
-    utils::tracing_span_assert::MultiNameExtractor<2>,
-> = once_cell::sync::Lazy::new(|| {
-    utils::tracing_span_assert::MultiNameExtractor::new("TenantId", ["tenant_id", "tenant"])
-});
+pub static TENANT_ID_EXTRACTOR: once_cell::sync::Lazy<MultiNameExtractor<2>> =
```
[the remark this answers is not in the document]
Comment (LizardWizzard) [reply pageserver/src/tenant/span.rs]: Done

File: pageserver/src/tenant/span.rs (hunk 3 above)
Comment (LizardWizzard) [reply pageserver/src/tenant/span.rs:2]: We dont have cfg-if anywhere, so just added one more debug_assertion cfg

**Review by koivunej — APPROVED** [review]

**Pull request merged**